test/integration/controllermanager/shoot/quota/quota_test.go

```diff
@@ -179,4 +178,16 @@ var _ = Describe("Shoot Quota controller tests", func() {
 			return testClient.Get(ctx, client.ObjectKeyFromObject(shoot), shoot)
 		}).Should(BeNotFoundError())
 	})
+
+	It("should consider longer (manually set) expiration times and delete the shoot", func() {
```

Comment:
This test succeeds even if the controller doesn't consider the prolonged expiration time and deletes the shoot immediately.
Maybe we should first check that it is not deleted after the original expiration time has passed before advancing the clock to after the new timestamp.

Reply:
Good catch :) Done in the last commit